src/cpp/session/modules/SessionRmdNotebook.R

```diff
@@ -398,15 +398,38 @@ assign(".rs.notebookVersion", envir = .rs.toolsEnv(), "1.0")
 
    # set up output_source
    outputOptions <- list(output_source = .rs.rnb.outputSource(rnbData))
-   
-   # call render with special format hooks
-   rmarkdown::render(input = inputFile,
-                     output_format = "html_notebook",
-                     output_options = outputOptions,
-                     output_file = outputFile,
-                     quiet = TRUE,
-                     envir = envir,
-                     encoding = encoding)
+
+   # knitr outputs relevant information in the form of messages that we attach to the error
+   renderMessages <- list()
+   tryCatch({
+      withCallingHandlers({
+         # call render with special format hooks
+         rmarkdown::render(input = inputFile,
+                           output_format = "html_notebook",
+                           output_options = outputOptions,
+                           output_file = outputFile,
+                           quiet = TRUE,
+                           envir = envir,
+                           encoding = encoding)
+      }, message = function(...) {
+         args <- list(...)
+         renderMessages <<- c(renderMessages, args[[1]])
+      })
+   }, error = function(e) {
+      messages <- list(e$message)
+
+      lapply(renderMessages, function(m) {
+        if (typeof(m) != "character") return();
+
+        result <- regexec("Quitting from lines ([0-9]+)-([0-9]+) ", text = m)
```

Comment:
do we need to capture more context from this line as is done here? https://github.com/rstudio/rstudio/blob/c89631fd312939edaa61b2e0315469e0104247de/src/cpp/session/modules/rmarkdown/SessionRMarkdown.cpp#L621